### DISCUSSION TITLE
BaseBranches not working for self-hosted

### DISCUSSION THREAD

**Discussion by chialiyun** [renovatebot/renovate#17215] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.118.0

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I wanted renovate to detect from a basebranch and create PRs with target to baseBranch instead of the default branch.

In the baseBranch (`test-branch`) I have already merged `renovate.json`. 

I have the following configuration:
```
module.exports = {
    branchPrefix: 'test-renovate-',
    onboardingBranch: 'test-renovate-onboarding',
    dryRun: false,
    username: 'renovate-release',
    gitAuthor: 'Renovate Bot <bot@renovateapp.com>',
    onboarding: false,
    platform: 'github',
    includeForks: true,
    dependencyDashboard: false,
    repositories: [
      {
        repository: 'owner/repo',
        baseBranch:'test-branch',
        draftPR: true,
      }
    ],
  };
```

renovate.json have the following content:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "draftPR": true
}
```

However, when selfhosted github actions runs, the repository is skipped. 
```
WARN: Config needs migrating
       "originalConfig": {
         "branchPrefix": "test-renovate-",
         "onboardingBranch": "test-renovate-onboarding",
         "dryRun": false,
         "username": "renovate-release",
         "gitAuthor": "Renovate Bot <bot@renovateapp.com>",
         "onboarding": false,
         "platform": "github",
         "includeForks": true,
         "dependencyDashboard": false,
         "repositories": [
           {
             "repository": "owner/repo",
             "baseBranch": "test-branch",
             "draftPR": true
           }
         ],
       },
       "migratedConfig": {
         "branchPrefix": "test-renovate-",
         "onboardingBranch": "test-renovate-onboarding",
         "dryRun": null,
         "username": "renovate-release",
         "gitAuthor": "Renovate Bot <bot@renovateapp.com>",
         "onboarding": false,
         "platform": "github",
         "includeForks": true,
         "dependencyDashboard": false,
         "repositories": [
           {
             "repository": "owner/repo",
             "baseBranches": ["test-branch"],
             "draftPR": true
           }
         ]
       }
```

```
 INFO: Repository is disabled - skipping 
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment, marked as answer]
Config must always be in the default branch of a repo, otherwise you must set `requireConfig=false`. If you want config to be read from base branches, you need to set `useBaseBranches=true` in addition to defining `baseBranches`.

Please don't raise any more bug issues, create a discussion first in future.
